airbyte-cdk/bulk/toolkits/load-dlq/src/main/kotlin/io/airbyte/cdk/load/write/dlq/DlqPipelineFactory.kt

```diff
@@ -0,0 +1,130 @@
+import io.airbyte.cdk.load.pipline.object_storage.ObjectLoaderPartFormatterStep
+import io.airbyte.cdk.load.pipline.object_storage.ObjectLoaderPartLoaderStep
+import io.airbyte.cdk.load.pipline.object_storage.ObjectLoaderUploadCompleterStep
+import io.airbyte.cdk.load.state.ReservationManager
+import io.airbyte.cdk.load.task.internal.LoadPipelineStepTaskFactory
+import io.airbyte.cdk.load.write.object_storage.ObjectLoader
+import io.micronaut.context.annotation.Factory
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+import java.io.OutputStream
+
+/** Factory to build a DeadLetterQueueLoadPipeline. */
+class DlqPipelineFactory(
+    private val dlqInputQueue: PartitionedQueue<PipelineEvent<StreamKey, DestinationRecordRaw>>,
+    private val dlqPipelineSteps: List<LoadPipelineStep>,
+    private val pipelineStepTaskFactory: LoadPipelineStepTaskFactory,
+    private val objectLoader: ObjectLoader,
+) {
+    fun <S : AutoCloseable> createPipeline(dlqLoader: DlqLoader<S>): LoadPipeline =
+        object :
+            LoadPipeline(
+                listOf(
+                    DlqLoaderPipelineStep(
+                        numWorkers = objectLoader.numPartWorkers,
+                        outputQueue = dlqInputQueue,
+                        taskFactory = pipelineStepTaskFactory,
+                        dlqLoader = dlqLoader,
+                    ),
+                    *dlqPipelineSteps.toTypedArray(),
+                )
+            ) {}
+}
+
+/** A Micronaut Factory to help initialize the component required for a DeadLetterQueuePipeline */
+@Factory
+class DlqPipelineFactoryFactory {
+    /** The end goal of this file. */
+    @Singleton
+    fun dlqPipelineFactory(
+        @Named("dlqInputQueue")
+        dlqInputQueue: PartitionedQueue<PipelineEvent<StreamKey, DestinationRecordRaw>>,
+        @Named("dlqPipelineSteps") dlqPipelineSteps: List<LoadPipelineStep>,
+        pipelineStepTaskFactory: LoadPipelineStepTaskFactory,
+        objectLoader: ObjectLoader,
+    ) = DlqPipelineFactory(dlqInputQueue, dlqPipelineSteps, pipelineStepTaskFactory, objectLoader)
+
+    /**
+     * This queue is the input queue of a "traditional" ObjectStorageLoadPipeline
+     *
+     * Effectively, a DLQ LoadPipeline is a custom pipeline step that may pass down records meant
+     * for the DLQ. The DLQ LoadPipeline upload to ObjectStorage leverages the usual object storage
+     * pipeline steps with the exception of the input.
+     *
+     * This Queue is the shim between the DlqLoaderStep and the ObjectStorage input.
+     */
+    @Named("dlqInputQueue")
+    @Singleton
+    fun dlqInputQueue(
+        @Named("globalMemoryManager") globalMemoryManager: ReservationManager
+    ): PartitionedQueue<PipelineEvent<StreamKey, DestinationRecordRaw>> =
+        ResourceReservingPartitionedQueue(
+            globalMemoryManager,
+            0.2,
+            1,
+            1,
+            100000,
```

Comment:
Nit: can we use named parameters here, so we can tell at a glance what these values map to?

Reply:
Good call, updated.